### PR TITLE
Add `--help-versions` option to miniconda component

### LINKED ISSUE #170: miniconda --help-versions
relates to #169 , but to help user more let's add this `--help-versions` which would list versions for the current OS/architecture from https://repo.anaconda.com/miniconda/

### PR DESCRIPTION
**Pull request by jwodder**
Closes #170.

### CODE REVIEW

File: src/datalad_installer.py [hunk 1]
```diff
@@ -87,6 +94,7 @@ def parse_log_level(level: str) -> int:
         return lv
 
 
+@dataclass
```
Comment (yarikoptic) [inline src/datalad_installer.py:97]: wondering for self-education: what is the motivation/need for using `@dataclass` for these classes now? The concise "Add --help-versions option to miniconda component" commit message gives no such information, making me wondering either it was just a generic refactoring to improve code "while at it" or needed to accomplish the drill ?

File: src/datalad_installer.py [hunk 2]
```diff
@@ -24,11 +24,17 @@
 from collections.abc import Callable, Iterator
 from contextlib import contextmanager, suppress
 import ctypes
+from dataclasses import InitVar, dataclass, field
+from email import policy
+from email.headerregistry import ContentTypeHeader
 from enum import Enum
 from functools import total_ordering
 from getopt import GetoptError, getopt
+from html.parser import HTMLParser
```
Comment (yarikoptic) [inline src/datalad_installer.py:33]: note to myself: this is a standard Python library - https://python.readthedocs.io/en/v2.7.2/library/htmlparser.html (new in 2.2!) so we remain true to the goal here of no external depends! cool

**Comment by yarikoptic** [comment]
looks and works great -- tested on "latest" - gave me one with python 3.11.5, and then on "py310_23.5.1-0" and got one with 3.10.12, cool, thank you!

**Comment by yarikoptic** [comment]
eh, was worth a `release` -- we should add a label to the next PR

File: src/datalad_installer.py (hunk 1 above)
Comment (jwodder) [reply src/datalad_installer.py:97]: It was just a generic refactoring.

**Pull request merged**